### PR TITLE
DAT 3856 - infobox builder theming validation

### PR DESCRIPTION
**Pull request by idradm**

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/services/Nodes/Node.class.php [hunk 1]
```diff
@@ -83,16 +83,23 @@ protected function getType() {
...
+				 || !$this->hasValidAttributeValues( $attribute, $value )
+			) {
 				return false;
 			}
 		}
 		return true;
 	}
 
+	protected function hasValidAttributeValues( $attribute, $value ) {
+		return empty( $this->allowedAttributes[ $attribute ] )
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/Nodes/Node.class.php]: What about '0' ?

File: extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php [hunk 2]
```diff
@@ -38,14 +38,4 @@ public function asJsonObject() {
 
 		return $data;
 	}
-
-	protected function hasValidAttributes() {
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/Nodes/NodeGroup.class.php:42]: Woohoo! :heart:

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ public function translateDataToMarkup( $builderData, $formatted = true ) {
 	 * @see PortableInfoboxBuilderServiceTest::translationsDataProvider
 	 */
 	public function translateMarkupToData( $infoboxMarkup ) {
-		$jsonObject = [ ];
+		$jsonObject = new stdClass();
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderService.class.php:33]: Good catch!

**Comment by sebastianmarzjan** [comment]
LGTM

**Pull request merged**